src/lib/AccountFinder.ts

```diff
@@ -0,0 +1,66 @@
+import { AccountInfo } from '@/lib/AccountInfo';
+import { NetworkClient } from '@nimiq/network-client';
+import { WalletStore } from '@/lib/WalletStore';
+
+export default class AccountFinder {
+    public static async findAccounts(
+        deriveAccounts: (startIndex: number, count: number) => Promise<Array<{ address: string, path: string }>>,
+        deriveWalletId: (firstAccount: { address: string, path: string }) => Promise<string>,
+        defaultLabel: string = 'Standard Account', // TODO move constant to a Constants class / file
+        onIntermediateResult?: (accounts: AccountInfo[]) => void,
+    ): Promise<AccountInfo[]> {
+        await NetworkClient.Instance.init(); // initialize NetworkClient if not initialized yet
+        const foundAccounts: AccountInfo[] = [];
+        let nextStartIndex = 0;
+        let derivedAccountsPromise = deriveAccounts(nextStartIndex, AccountFinder.MAX_ALLOWED_GAP);
+        nextStartIndex += AccountFinder.MAX_ALLOWED_GAP;
+
+        // calculate walletId (potentially using the first account) and read existing accounts
+        const existingAccountsPromise = derivedAccountsPromise
+            .then((derivedAccounts) => deriveWalletId(derivedAccounts[0]))
+            .then((walletId) => WalletStore.Instance.get(walletId))
+            .then((walletInfo) => walletInfo ? walletInfo.accounts : new Map<string, AccountInfo>());
```

Comment:
I still think this can be a callback to the user of this library, which would save us a database access for all Keyguard users.

Reply:
indeed. This code has now also changed though